geomstats/geometry/discrete_curves.py

```diff
@@ -401,13 +413,13 @@ def inner_product(self, tangent_vec_a, tangent_vec_b, curve):
                 "is only implemented for discrete curves "
                 "embedded in a Euclidean space."
             )
-        d_srv_vec_a = self.aux_differential_square_root_velocity(tangent_vec_a, curve)
-        d_srv_vec_b = self.aux_differential_square_root_velocity(tangent_vec_b, curve)
-        inner_prod = self.srv_inner_product(d_srv_vec_a, d_srv_vec_b)
+        d_srv_vec_a = self.aux_differential_srv_transform(tangent_vec_a, curve)
+        d_srv_vec_b = self.aux_differential_srv_transform(tangent_vec_b, curve)
+        inner_prod = self._srv_inner_product(d_srv_vec_a, d_srv_vec_b)
 
         return inner_prod
 
-    def srv_inner_product(self, srv_1, srv_2):
+    def _srv_inner_product(self, srv_1, srv_2):
```

Comment:
should this be called "inner_product"? so that the user writes: `srv_metric.inner_product`?